src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -492,6 +557,81 @@ private void StartWriting(byte[] buffer, int offset, int count, AsyncProtocolReq
             }
         }
 
+        private async Task StartWritingAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            // We loop to this method from the callback.
```

Comment:
The main bit that is essentially different is a call to `WriteAsync` vs `Write` etc, and returning a Task. However, trying to extract the methods made it messy with parameters. Went for a quick fix, rather than something I'm proud of 😢 It can be made much better; though needs to support three forms of Write...